server/modules/monitor/account_mon.c

```diff
@@ -231,6 +233,13 @@ static void *startMonitor(void *arg, void *opt) {
                         strcpy(handle->topic_name, params->value);
                 }
 
+                if(strcasecmp(params->name, "service") == 0) {
```

Comment:
It kinda sucks the account monitor needs to know the top level service, but I don't know if there's a better way.
It's checked in [check_db_name_after_auth](https://github.com/zendesk/MaxScale/blob/sd/fix-db-select/server/modules/protocol/mysql_common.c#L2179) from mysql_client.c.